project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/model/IJDDItem.kt

```diff
@@ -75,4 +82,19 @@ data class PsiWithBodyDDItem(
                 )
             }
     }
+
+    interface PsiWithBodyTransformer<T> {
+        fun transform(classInitializer: KtClassInitializer): T
+        fun transform(function: KtNamedFunction): T
+        fun transform(lambdaExpression: KtLambdaExpression): T
+        fun transform(accessor: KtPropertyAccessor): T
+
+        fun transform(element: PsiElement): T = when (element) {
+            is KtClassInitializer -> transform(element)
+            is KtNamedFunction -> transform(element)
+            is KtLambdaExpression -> transform(element)
+            is KtPropertyAccessor -> transform(element)
+            else -> error("Invalid PSI element type: ${element::class.simpleName}. Expected one of: KtClassInitializer, KtNamedFunction, KtLambdaExpression, KtPropertyAccessor")
+        }
+    }
```

Comment:
Well, I quite don't understand the reasoning behind making this abstraction. I mean sure it makes sense to mark all similar classes, but why not just give up on it before it's actually needed?

I mean it's only two implementations: `PsiBodyReplacer` and `PsiBodyTypeRenderer` that share the same "affected" psi types. However, if there is another set of psis that could be transformed (I look on instance-level), then we would need some tweaking and workarounds. 

That's the same problem as when I force `PsiDDItem` to be constructable from chosen set of Psi elements

Reply:
I need this interface only for the body-replacement algorithm. If we change the type of elements with the bodies we are handling, we will change this interface and remember to change all parts based on the handling types. I'd rather prefer also to make the PsiWithBodyDDItem as some hierarchy, but I'm afraid that your PR will make some other changes, so I tried to keep the current version for seamless integration

Reply:
But if you want I can move this interface to the other place

Reply:
Nah, that's fine :)